src/Exception/InvalidIfResultException.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sofascore\PurgatoryBundle\Exception;
+
+final class InvalidIfResultException extends RuntimeException
+{
+    private const MESSAGE = 'Expected return value of "if" expression (%s) to be boolean, got %s';
+
+    public function __construct(
+        public readonly string $expression,
+        public readonly mixed $result,
+    ) {
+        parent::__construct(\sprintf(self::MESSAGE, $expression, \gettype($result)));
```

Comment:
@Brajk19 Can we use [`get_debug_type`](https://www.php.net/manual/en/function.get-debug-type.php) here?